app/src/main/java/com/github/se/icebreakrr/model/message/MeetingRequestViewModel.kt

```diff
@@ -95,19 +84,74 @@ class MeetingRequestViewModel(
               "targetToken" to meetingRequestState.targetToken,
               "senderUID" to meetingRequestState.senderUID,
               "body" to MeetingRequestManager.ourName + " : " + meetingRequestState.message,
-              "picture" to meetingRequestState.picture,
-              "location" to meetingRequestState.location)
+          )
       try {
         val result =
             functions
                 .getHttpsCallable(SEND_MESSAGE_FUNCTION_NAME) // Cloud Function name
                 .call(data)
                 .await()
 
-        meetingRequestState = meetingRequestState.copy(message = "", picture = null)
+        meetingRequestState = meetingRequestState.copy(message = "")
       } catch (e: Exception) {
         Log.e("FIREBASE ERROR", "Error sending message", e)
       }
     }
   }
+
+  fun sendMeetingRequest() {
+    viewModelScope.launch {
+      val data =
+          hashMapOf(
+              "targetToken" to meetingRequestState.targetToken,
+              "senderUID" to meetingRequestState.senderUID,
+              "message" to MeetingRequestManager.ourName + " : " + meetingRequestState.message,
+          )
+      try {
+        val result =
+            functions
+                .getHttpsCallable(SEND_MEETING_REQUEST) // Cloud Function name
+                .call(data)
+                .await()
+
+        meetingRequestState = meetingRequestState.copy(message = "")
+      } catch (e: Exception) {
+        Log.e("FIREBASE ERROR", "Error sending message", e)
+      }
+    }
+  }
+
+  fun addToMeetingRequestSent(receiverUID: String) {
```

Comment:
Nitpick: Methods like addToMeetingRequestSent and addToMeetingRequestInbox use profilesViewModel.getProfileByUidAndThen, which relies on the selected profile’s value being set. This indirect dependency could introduce race conditions if the value is updated elsewhere.